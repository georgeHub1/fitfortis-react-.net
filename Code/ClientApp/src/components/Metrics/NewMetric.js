import * as React from 'react';
import { Button, Icon, Input, Drawer } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import styles from './styles.module.less';
import SelectedMetric from '../../containers/Metrics/SelectedMetric';
import PropTypes from 'prop-types';
import AvailableMetric from './AvailableMetric';
import { MetricsChartDataProps } from '../../containers/Metrics/metrics.chart.propTypes';

class NewMetric extends React.Component {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    intl: PropTypes.shape({
      formatMessage: PropTypes.func.isRequired
    }).isRequired,
    selectedMetrics: PropTypes.arrayOf(MetricsChartDataProps).isRequired,
    availableMetrics: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired
      })
    ).isRequired,
    handleSearch: PropTypes.func.isRequired,
    countAvailable: PropTypes.number.isRequired,
    chartId: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    addMetricToChart: PropTypes.func.isRequired
  };

  isUnitMetircs = metric => {
    return !metric.canBeDisplayedWithOther;
  }
  render () {
    const {
      isOpen: open,
      onClose: close,
      intl: { formatMessage },
      selectedMetrics,
      availableMetrics,
      value,
      handleSearch,
      countAvailable,
      chartId,
      addMetricToChart
    } = this.props;

    return (
      <Drawer
        className="drawerComponentSymptomChecker"
        title={
          <div className={styles.header}>
            {<FormattedMessage id="NewMetric.header" />}
          </div>
        }
        placement="right"
        closable={false}
        onClose={close}
        visible={open}
      >
        <div className={`maincardWrapperNewMetric ${styles.cardWrapperNewMetric}`}>
          <div className={styles.selectedBlock}>
            <div className={`mainselectedBlock ${styles.selectedMetrics}`}>
              {selectedMetrics.length > 0 ? (
                selectedMetrics.map(item => <SelectedMetric
                   key={item.chartMetricId}
                    chartMetricId={item.chartMetricId}
                  chartId={chartId}
                     chartMetric={item} />)
              ) : (
                  <div>
                    <FormattedMessage id="NewMetric.chooseOne" />
                  </div>
                )}
            </div>
          </div>
          <div className={styles.availableBlock}>
            <div className={styles.searchBlock}>
              <Input
                value={value}
                onChange={handleSearch}
                className={styles.searchInput}
                prefix={<Icon type="search" className={styles.iconSearch} />}
                placeholder={formatMessage({ id: 'NewMetric.searchPlaceholder' })}
              />
            </div>
            {countAvailable === 1 ? (
              <div className={styles.quantityAvailable}>
                {countAvailable}
                &nbsp;
                <FormattedMessage id="NewMetric.availableMetric" />
              </div>
            ) : (
                <div className={styles.quantityAvailable}>
                  {countAvailable}
                  &nbsp;
                <FormattedMessage id="NewMetric.availableMetrics" />
                </div>
              )}
            <div className={styles.availableItems}>
              {availableMetrics.map(item => <AvailableMetric
                key={item.id}
                metricId={item.id}
                chartId={chartId}
                searchValue={value}
                metric={item}
                isHideUintMetric={this.isUnitMetircs(item)}
                onAdd={addMetricToChart}
              />)}
            </div>
          </div>
        </div>
        <div className={`Metrics_bottom_btn ${styles.btnBlock}`}>
          <Button onClick={close}>
            <FormattedMessage id="NewMetric.closeBtn" />
          </Button>
        </div>
      </Drawer>
    );
  }
}
export default injectIntl(NewMetric);
