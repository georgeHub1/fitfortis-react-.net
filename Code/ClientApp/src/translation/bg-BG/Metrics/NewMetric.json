{
    "header": "Показатели",
    "deleteBtn": "Махни",
    "closeBtn": "Затвори",
    "addBtn": "Добави",
    "searchPlaceholder": "Търси",
    "metricSettings": "Настройки",
    "chooseOne": "Избраните показатели ще се появят тук. Използвай полето по-долу за да търсиш и да добавяш показатели.",
    "availableMetric": "наличен показател",
    "availableMetrics": "налични показатели"
}