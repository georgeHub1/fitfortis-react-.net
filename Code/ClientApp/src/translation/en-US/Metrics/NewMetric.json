{
    "header": "Metrics",
    "deleteBtn": "remove",
    "closeBtn": "Close",
    "addBtn": "Add",
    "searchPlaceholder": "Search for metric",
    "metricSettings": "Metric Settings",
    "chooseOne": "Your selection will appear here. Use the field below to search for available metrics.",
    "availableMetric": "available metric",
    "availableMetrics": "available metrics"
}