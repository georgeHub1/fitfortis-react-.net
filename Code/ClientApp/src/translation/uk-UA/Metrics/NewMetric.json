{
    "header": "Додати Метрику",
    "deleteBtn": "Видалити",
    "closeBtn": "Закрити",
    "addBtn": "Додати",
    "searchPlaceholder": "Пошук по метриках",
    "metricSettings": "Параметри метрики",
    "chooseOne": "Ваш вибір з'явиться тут. Використовуйте поле нижче, щоб знайти доступні діаграми.",
    "availableMetric": "доступні метрик",
    "availableMetrics": "доступні метрики"
}