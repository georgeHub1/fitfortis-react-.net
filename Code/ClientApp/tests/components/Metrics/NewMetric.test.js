import React from 'react';
import renderer from 'react-test-renderer';
import NewMetric from '../../../src/components/Metrics/NewMetric.js';

const Container = NewMetric;

describe('NewMetric', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Container
        isOpen={false}
        onClose={jest.fn()}
        intl={{ formatMessage: jest.fn() }}
        selectedMetrics={[]}
        availableMetrics={[]}
        handleSearch={jest.fn()}
        countAvailable={0}
        chartId={""}
        value={""}
        addMetricToChart={jest.fn()}
      />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
