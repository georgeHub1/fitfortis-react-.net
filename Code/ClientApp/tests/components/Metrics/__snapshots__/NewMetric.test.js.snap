// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMetric renders correctly 1`] = `null`;
